diff --git a/plugins/notifications/notificationsplugin.cpp b/plugins/notifications/notificationsplugin.cpp
--- a/plugins/notifications/notificationsplugin.cpp
+++ b/plugins/notifications/notificationsplugin.cpp
@@ -11,7 +11,7 @@
 #include <dbushelper.h>
 
 #include <KPluginFactory>
-#if !defined(Q_OS_WIN) && !defined(Q_OS_MAC)
+#if !defined(Q_OS_WIN) && !defined(Q_OS_MAC) && defined(KDECONNECT_NOTIFICATIONS_X11)
 #include <KStartupInfo>
 #include <private/qtx11extras_p.h>
 #endif
@@ -152,12 +152,6 @@ void NotificationsPlugin::replyRequested(Notification *noti)
     SendReplyDialog *dialog = new SendReplyDialog(originalMessage, replyId, appName);
     connect(dialog, &SendReplyDialog::sendReply, this, &NotificationsPlugin::sendReply);
     dialog->show();
-#if !defined(Q_OS_WIN) && !defined(Q_OS_MAC)
-    auto window = qobject_cast<QWindow *>(dialog->windowHandle());
-    if (window && QX11Info::isPlatformX11()) {
-        KStartupInfo::setNewStartupId(window, QX11Info::nextStartupId());
-    }
-#endif
     dialog->raise();
 }
