diff --git a/src/kded/helpers/requestpin.cpp b/src/kded/helpers/requestpin.cpp
--- a/src/kded/helpers/requestpin.cpp
+++ b/src/kded/helpers/requestpin.cpp
@@ -19,7 +19,6 @@
 #include <KLocalizedString>
 #include <KNotification>
 #include <KWindowSystem>
-#include <KX11Extras>
 
 RequestPin::RequestPin(BluezQt::DevicePtr device, bool numeric, QObject *parent)
     : QObject(parent)
@@ -89,9 +88,6 @@
 
     dialog->show();
 
-    if (KWindowSystem::isPlatformX11()) {
-        KX11Extras::forceActiveWindow(dialog->winId());
-    }
 }
 
 void RequestPin::checkPin(const QString &pin)
